### PR TITLE
csvfile - add a keycol parameter to specify in which column to search.

### PR DESCRIPTION
**Pull request by helmo**
##### SUMMARY

The csvfile lookup module is great but it was limited to searching in the first column. This PR adds a `keycol` parameter (default's to the first) to be able to look in the n'th column.

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: lib/ansible/plugins/lookup/csvfile.py [hunk 1]
```diff
@@ -16,6 +16,10 @@
       col:
         description:  column to return (0 indexed).
         default: "1"
+      keycol:
+        description:  column to search in (0 indexed).
+        default: "0"
+        version_added: "2.17"
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/csvfile.py:23]: ```type: int```, remove quotes from default

File: lib/ansible/plugins/lookup/csvfile.py [hunk 2]
```diff
@@ -122,14 +126,14 @@ def __iter__(self):
...
-    def read_csv(self, filename, key, delimiter, encoding='utf-8', dflt=None, col=1):
+    def read_csv(self, filename, key, delimiter, encoding='utf-8', dflt=None, col=1, keycol=0):
 
         try:
             f = open(to_bytes(filename), 'rb')
             creader = CSVReader(f, delimiter=to_native(delimiter), encoding=encoding)
 
             for row in creader:
-                if len(row) and row[0] == key:
+                if len(row) and row[int(keycol)] == key:
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/csvfile.py]: don't need to convert here if the option was set and used correctly.

**Review by bcoca — CHANGES_REQUESTED** [review]
I understand this was mostly copying the existing `col` but that field was incorrectly used as a string and used manual typecasting .. while the config system already can do this for you.

note: we should update col as bugfix in another PR

File: lib/ansible/plugins/lookup/csvfile.py (hunk 1 above)
Comment (helmo) [reply lib/ansible/plugins/lookup/csvfile.py:23]: Thanks, now added

**Review by bcoca — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

**Pull request merged**